bindings/nodejs/src/lib.rs

```diff
@@ -15,6 +15,8 @@
 // specific language governing permissions and limitations
 // under the License.
 
+#![allow(clippy::not_unsafe_ptr_arg_deref)]
```

Comment:
Please mark them as unsafe instead of allow clippy.

Reply:
This unsafe warning is from a macro [here](https://github.com/apache/incubator-opendal/pull/3484/files#diff-ca4184f1ae4e7a2a05946c3af8bceb7ecb303597fa60ed20047820cacb707b85R734). I don't how to mark a macro as unsafe. And seems that it's impossible to mark a macro as unsafe now.
Marking the function as unsafe doesn't work. 🥲

Reply:
Maybe we can use `#[napi(constructor)]` instead like what we do for Operator?

Reply:
The `napi` macro will make sure every parameter and return value's type implements `ToNapiValue`, but a pointer (the Box<dyn layer>)can't implement it. To pass a pointer to js env, it’s necessary to use this compat mode, the `js_function`.

Reply:
I got the problem now.